packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/mockapi.ts

```diff
@@ -155,6 +173,141 @@ Scenarios.Azure_ResourceManager_Models_Resources_SingletonTrackedResources_listB
   ),
 ]);
 
+Scenarios.Azure_ResourceManager_Models_Resources_ExtensionsResources_get = passOnSuccess([
+  mockapi.get(
+    "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/topLevelTrackedResources/:topLevelResourceName/providers/Azure.ResourceManager.Models.Resources/extensionsResources/:extensionName",
```

Comment:
Will this path work?

Reply:
Yes, the paths of  `ExtensionResource` in `mockapi.tsp` has been verified and works fine.

Reply:
Thanks, this means our code doesn't have encoding issue with `resourceUri`, is that right?

Another thing is, where did you get the value to pass into the `resourceUri`? Seems it lacks namespace `Azure.ResourceManager.Models.Resources` for topLevelTrackedResource...

Reply:
@v-hongli1 check the top level resource id: https://github.com/Azure/cadl-ranch/blob/4022f01c7cebed9df15ba36d591fb5784a6b23ff/packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/toplevel.tsp#L48

The one you passed in is wrong.

Reply:
1. First, resourceUri still has encoding issues
2. Secondly, the top level resource id is also passed through resourceUri.
![image](https://github.com/user-attachments/assets/3f27ea42-f6da-4c9e-bad4-04ae2c8f76a4)

Reply:
Got it.

For 1, we should wait for the fix before merging this PR.

For 2, The value is not correct. It should be: https://github.com/Azure/cadl-ranch/blob/4022f01c7cebed9df15ba36d591fb5784a6b23ff/packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/toplevel.tsp#L48

Reply:
@XiaofeiCao 
Use `@parentResource` or `@resource("topLevelTrackedResources/top/extensionsResources")`?

Reply:
No. I mean the one in your mockapi.ts(same as the one you passed to `.withExistingResourceUri()` in your `testExtensionResource()`) is wrong.

Use `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Models.Resources/topLevelTrackedResources/top`.

Reply:
Fixed by commit https://github.com/Azure/cadl-ranch/pull/701/commits/7f74cde43b5bb7830af8575e11e3d5b9743c57c6

Reply:
@v-hongli1 , after I offline discussed with chenjie, we should make mockapi return double "//".